RadarSDK/RadarAPIClient.m

```diff
@@ -251,10 +251,9 @@ - (void)trackWithLocation:(CLLocation *_Nonnull)location
         params[@"floorLevel"] = @(location.floor.level);
     }
     long nowMs = (long)([NSDate date].timeIntervalSince1970 * 1000);
-    if (!foreground) {
-        long timeInMs = (long)(location.timestamp.timeIntervalSince1970 * 1000);
-        params[@"updatedAtMsDiff"] = @(nowMs - timeInMs);
-    }
+    long locationMs = (long)(location.timestamp.timeIntervalSince1970 * 1000);
```

Comment:
ok so I guess the approach now is:
• change `foreground = true` to also use `updatedAtMsDiff` (for consistency?  I might have been the one to bring this up?)
• Don't special case `verified = true` on the client side, handle it differently on the server side

If we decided not to change `foreground = true` to `updatedAtMsDiff`, would the other changes to `updatedAtMsDiff` in these PRs be enough to fix the bug?  Or is the inconsistent behavior between `foreground = true` and `foreground = false` causing a problem?

Reply:
>  change foreground = true to also use updatedAtMsDiff (for consistency? I might have been the one to bring this up?)

Yes, for three reasons:
- not using `updatedAtMsDiff` causes a problem when a foreground location gets forced to server time while background locations are adjusted per `updatedAtMsDiff`, leading to inconsistent ordering (a subsequent background location can be brought erroneously before the foreground location by the `updatedAtMsDiff` adjustment).
- for consistency
- because there's no empirical reason to specially trust foreground locations from the OS as time-correct. It's true that foreground locations aren't subject to delays related to backgrounding, but the OS can and does deliver locations when foregrounded that have a non `now` timestamp

Reply:
> • Don't special case `verified = true` on the client side, handle it differently on the server side

Correct, per slack conversation. I'll link in #mobile